File: app/backend/app/schemas/divergence.py
from pydantic import BaseModel


class CityPriceDivergence(BaseModel):
    city_code: str
    city_name: str
    prefecture_name: str
    year: int
    avg_published_price: int
    avg_transaction_price: int
    divergence_rate: float
    published_count: int
    transaction_count: int


class TownPriceDivergence(BaseModel):
    city_code: str
    city_name: str
    prefecture_name: str
    district_name: str
    year: int
    avg_published_price: int
    avg_transaction_price: int
    divergence_rate: float
    published_count: int
    transaction_count: int

File: app/frontend/src/types/divergence.ts
export interface CityPriceDivergence {
  city_code: string;
  city_name: string;
  prefecture_name: string;
  year: number;
  avg_published_price: number;
  avg_transaction_price: number;
  divergence_rate: number; // 正 = 取引価格 > 公示地価（%）
  published_count: number;
  transaction_count: number;
}

export interface TownPriceDivergence {
  city_code: string;
  city_name: string;
  prefecture_name: string;
  district_name: string;
  year: number;
  avg_published_price: number;
  avg_transaction_price: number;
  divergence_rate: number;
  published_count: number;
  transaction_count: number;
}
